### PR TITLE
Replace `created/upserted/absent` with `present/absent` & simplify `generate_value`

### PR DESCRIPTION
**Pull request by verkaufer**
This PR simplifies how users manage the state of items and generated values when running a playbook. It addresses #10 and #11 

Users can now specify an item's state as either `state: present` or `state: absent`. Additionally, the `generate_value` option on item fields now uses `always/never/on_create` instead of `true/false`.

## Generated values

Previously, users could only toggle whether a field is generated server-side using `true/false`. This was an issue if you only wanted to generate the value _once_ and then persist it across subsequent playbook executions. We originally implemented `overwrite: true/false` as an option on each field to mitigate this, but users found it confusing. 

Going forward, users can supply `always/never/on_create` for the `generate_value` field option:
- `always` will regenerate the field value every time the playbook is executed. It instructs the playbook to ignore the `value` option, if defined.
- `never` means the field's `value` option will be used and the field will never be generated server-side.
- `on_create` will generate the field **once** and then persist that field's value across subsequent playbook execution.
   - The field must have the same field label across executions.

## Item State

Because we no longer use the `overwrite` option, I decided to simplify the item state option as well. We now follow the Ansible convention of `state: present/absent`. If the item state is `present` and an API request to find an item with the same name returns 0 results, the item is created.  If the state is `absent`, the item is deleted.

****Note**: changing the name of the item requires the user to supply the item UUID. If the UUID is not provided, a new item with the new name is created and the old item is not deleted.**

### CODE REVIEW

File: tests/unit/plugins/module_utils/test_protected_item_fields.py [hunk 1]
```diff
@@ -1,142 +1,142 @@
...
-        fieldset_from_server,
-        playbook_fieldset_with_unprotected_field
-    )
-
-    result_field = _get_field_by_label(unprotected_field["label"], updated_fieldset)
-    assert result_field["value"] == unprotected_field["value"]
-
-    protected_field = _get_field_by_label(FIELD_FROM_SERVER_LABEL, updated_fieldset)
-    assert protected_field["value"] == FIELD_FROM_SERVER["value"]
+# from __future__ import (absolute_import, division, print_function)
```
Comment (LeofkJ) [inline tests/unit/plugins/module_utils/test_protected_item_fields.py:1]: Is this file meant to be commented out?

**Review by LeofkJ — COMMENTED** [review]
Code looks good, I'll run the tests soon

File: tests/unit/plugins/module_utils/test_protected_item_fields.py (hunk 1 above)
Comment (verkaufer) [reply tests/unit/plugins/module_utils/test_protected_item_fields.py:1]: Whoops, meant to delete this file. Thanks for catching!

**Review by LeofkJ — APPROVED** [review]
Changes look good :+1:

**Review by LeofkJ — APPROVED** [review]
Every test pass, approved!

**Pull request merged**